pkg/server/server.go

```diff
@@ -27,10 +28,16 @@ func Serve(ctx context.Context, config ekcoops.Config, client *cluster.Controlle
 			}
 			return
 		}
-
-		w.WriteHeader(http.StatusOK)
-		_, err = w.Write([]byte(message))
+		data, err := json.Marshal(status)
 		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			if _, err := w.Write([]byte(err.Error())); err != nil {
+				log.Printf("marshaling json: %v", err)
```

Comment:
this is the error from the call to w.Write, not the json marshalling error

```suggestion
				log.Printf("write json marshaling error: %v", err)
```

Reply:
Good catch.